### PR TITLE
Enable default AWS S3 SSE

### PR DESCRIPTION
**Pull request by gabrieljackson**
This sets the default encryption scheme for new installation S3
buckets to use AES-256 encryption. The Mattermost filestore SSE
configuration option is also set to true to force the behavior
when a license is present.

This also corrects an issue where the cloud installation ID env
var was lost when updating installations.

Fixes https://mattermost.atlassian.net/browse/MM-24783

```release-note
Enable default AWS S3 SSE
```

### CODE REVIEW

File: internal/provisioner/kops_provisioner_cluster_installation.go
Comment (cpanato) [inline internal/provisioner/kops_provisioner_cluster_installation.go:533]: minio supports the same encryption as AWS

Comment (gabrieljackson) [reply internal/provisioner/kops_provisioner_cluster_installation.go:533]: I tried enabling encryption with Minio and the Mattermost server started throwing errors and I couldn't connect. Is there maybe something that needs to be set on the Minio side?

Comment (cpanato) [reply internal/provisioner/kops_provisioner_cluster_installation.go:533]: yes, you need to configure minio to accept that as well, but I don't know if the operator supports that, need to check the options

Comment (gabrieljackson) [reply internal/provisioner/kops_provisioner_cluster_installation.go:533]: Okay, I figured that may be the case. Either way, I propose that we merge this as is and update it later if we can add the operator option to support it.

File: internal/tools/aws/s3.go
Comment (cpanato) [inline internal/tools/aws/s3.go:38]: we might need to add this permission in the provisioner user
```
s3:PutEncryptionConfiguration
```

Comment (cpanato) [reply internal/tools/aws/s3.go:38]: I will check if we need

for now lets /hold this and not merge

Comment (gabrieljackson) [reply internal/tools/aws/s3.go:38]: Thank you for checking that!

Comment (cpanato) [reply internal/tools/aws/s3.go:38]: need to merge and update the envs first: https://github.com/mattermost/mattermost-cloud-monitoring/pull/161

**Review by cpanato — APPROVED** [review]

**Comment by gabrieljackson** [comment]
Just talked to @cpanato and the required permissions are now deployed. Please review when you have time @gigawhitlocks.

**Pull request merged**
